internal/sign/job/signer.go

```diff
@@ -66,6 +69,24 @@ func (m *signer) MakeJobTemplate(mod kmmv1beta1.Module, signConfig *kmmv1beta1.S
 		volumeMounts = append(volumeMounts, utils.MakeSecretVolumeMount(mod.Spec.ImageRepoSecret, "/docker_config"))
 	}
 
+	if signConfig.UnsignedImageRegistryTLS.Insecure {
+		args = append(args, "--insecure-pull")
+	}
+
+	if signConfig.UnsignedImageRegistryTLS.InsecureSkipTLSVerify {
+		args = append(args, "--skip-tls-verify-pull")
+	}
+
+	if registryTLS != nil {
```

Comment:
this question is more related to a previous PR: 
why not RegistryTLS in kernel mapping needs to be a pointer? Why not make is a struct, and then we don't need to check if the pointer is not nil. In case user does not set it, then  by default its fields are false

Reply:
This is a good point. To be honest I don't know what is the best practice here? I was following what we have in the `build` section.

One of the benefits of using a pointer is that we can just send `nil` in all the unit tests that doesn't care about this part, and then we can test only that part instead of forcing all previous tests to adapt.

When writing tests, I am happy that I can just send `nil`, when looking at the code then we could have indeed remove that `nil` check.

I am not sure which option is better, but I lean toward having `nil` if not specified.

Reply:
I always prefer to prioritize the production code over tests, especially unit-test.  In the unit-test itself, you can always leave that part out, and it will be initialized by default to false.

Reply:
Fixed. You can see the changes in the last commit. Once https://github.com/kubernetes-sigs/kernel-module-management/pull/163 (which is a blocker for this PR) is merged, I will rebase and squash.